### ISSUE TITLE
Building hono-core with 'embed-dependencies' maven profile fails

### ISSUE THREAD

**Issue by calohmn** [eclipse/hono#3069]
As stated in https://github.com/eclipse/hono/pull/3063#discussion_r802289717:

In Hono 1.12, building hono-core with the 'embed-dependencies' maven profile for OSGi support fails with this error:
```
[INFO] --- maven-bundle-plugin:3.5.0:bundle (default-bundle) @ hono-core ---
[WARNING] Bundle org.eclipse.hono:hono-core:bundle:1.12.0 : No translation found for macro: opentracing.version
[WARNING] Bundle org.eclipse.hono:hono-core:bundle:1.12.0 : No translation found for macro: range;[==,+)
[ERROR] Bundle org.eclipse.hono:hono-core:bundle:1.12.0 : Invalid syntax for version: ${opentracing.version}, for cmd: range, arguments; [range, [==,+)]
[ERROR] Error(s) found in bundle configuration
```

Workaround is to use `-Dopentracing.version=0.33.0` in the maven invocation.

Relates to #2935.

**Comment by sophokles73** [comment]
FMPOV we can close this as *won't fix* ...

**Comment by calohmn** [comment]
I think we can at least fix the maven property reference in 1.12.x (see #3071) and then remove the feature for 2.0.0.
